library/src/main/java/com/novoda/downloadmanager/lib/BatchStatusRepository.java

```diff
@@ -61,14 +67,6 @@ int getBatchStatus(long batchId) {
             }
         }
 
-        // check for error statuses first
-        for (int i = 0; i < statusCounts.size(); i++) {
-            int statusCode = statusCounts.keyAt(i);
-            if (Downloads.Impl.isStatusError(statusCode) && statusCounts.get(statusCode) > 0) {
```

Comment:
At this point, `statusCounts.get(status)` is for sure `>0` because of https://github.com/novoda/download-manager/pull/50/files#diff-45bc1abd9ac053b56014cfacfee3f317L56 and so this check is unnecessary